packages/ai/src/authorizers/types.ts

```diff
@@ -15,25 +15,62 @@ export type ToolWithAuthHandler<I, O, C> = (
 
 export const Auth0ClientSchema = z.object({
   domain: z.string().default(() => process.env.AUTH0_DOMAIN!),
-  clientId: z.string().default(() => process.env.AUTH0_CLIENT_ID!),
+  clientId: z.string().optional(),
   clientSecret: z
     .union([z.string(), z.undefined()])
     .transform((v) => v ?? process.env.AUTH0_CLIENT_SECRET),
+  resourceServerClientId: z
+    .union([z.string(), z.undefined()])
+    .transform((v) => v ?? process.env.LINKED_CLIENT_ID),
+  resourceServerClientSecret: z
+    .union([z.string(), z.undefined()])
+    .transform((v) => v ?? process.env.LINKED_CLIENT_SECRET),
   telemetry: z.boolean().optional(),
   clientInfo: z
     .object({
       name: z.string(),
     })
     .passthrough()
     .optional(),
+}).refine((data) => {
+  // Either clientId OR resourceServerClientId must be provided
+  return data.clientId || data.resourceServerClientId;
+}, {
+  message: "Either clientId or resourceServerClientId must be provided",
```

Comment:
Sort of uncertain about this change. It seems that the API will not require a `clientId` if we pursue this flow as described (having the API server perform the token exchange), however we'd need to think more deeply here now around impact to existing flows when the `clientId` is not present.

Reply:
updated in 3b4069f8c340f8d39e1133d67e9da979be2fbd41